tmpl/.gitignore.tmpl

```diff
@@ -1,2 +1,4 @@
-*
-.*
\ No newline at end of file
+gen/
+profile*
+nix-profile/
+shell_history
```

Comment:
alternatively, this could be changed to:
```
*
.*
```
and later if adding features we want checked in, then we can modify this file as:
```
*
.*
!<feature>
```
https://stackoverflow.com/a/5250314

Reply:
discussed IRL, and we'll move to this way.